docs/IntegrationGuide.md

```diff
@@ -1,95 +1,306 @@
+        ```
 
-  - create *`payment-paypal`* custom type with fields, specified in 
-  [ctPaymentCustomType.json](/src/main/resources/referenceModels/ctPaymentCustomType.json). See also 
-  [CtpPaymentCustomFields](/src/main/java/com/commercetools/payment/constants/ctp/CtpPaymentCustomFields.java) class.
+    1. Backend extracts PayPal specific parameters: `paymentId`, `PayerID` and POSTs them in the request body to Paypal-integration (see also [HTTP Responses](#http-responses)) for payment execution. Example:
+    ```
+    POST https://paypal-plus-integration-server.com/${tenantName}/commercetools/execute/payments/
+    {"paypalPlusPaymentId": "${paymentId}", "paypalPlusPayerId": "${payerId}"}
+    ```
+    1. In case of **invoice payment**, the bank details for the invoice will be saved as custom fields in the Payment object. Example:
+    ```json
+    {
+       "custom": {
+        "type": {
+          "typeId": "type",
+          "id": "1455d4e6-41b4-yyyy-xxxx-4f120864e231"
+        },
+        "fields": {
+          "reference": "6KF07542JV235932C",
+          "paymentDueDate": "2017-09-27",
+          "amount": {
+            "centAmount": 200,
+            "currencyCode": "EUR"
+          },
+          "paidToIBAN": "DE1212121212123456789",
+          "paidToAccountBankName": "Deutsche Bank",
+          "paidToAccountHolderName": "PayPal Europe",
+          "paidToBIC": "DEUTDEDBPAL"
+        }
+      }
+    }
+    ```
 
-## Front-end workflow
+### Validations
 
-### Prevent shipping (delivery) address change in Paypal Payment Dialog
+#### Validate cart state
+Check if current cart has been ordered already (`Cart#cartState` = `Ordered`). In this case load order by ordered cart ID and show oder confirmation page. This might happen if cart has been already ordered in a different tab or by asynchronous process like [commercetools-payment-to-order-processor](https://github.com/commercetools/commercetools-payment-to-order-processor) job.
+
+#### Recalculate cart
+To ensure cart totals are always up-to-date execute cart [recalculate](https://dev.commercetools.com/http-api-projects-carts.html#recalculate). Time limited discounts are not removed/invalidated from cart automatically. They are validated on recalculate and order creation only.
+
+#### Validate payment amount
+There must be at least one CTP payment object of type PayPal Plus (`#Payment#paymentMethodInfo#paymentInterface` = `PAYPAL_PLUS`) where `Payment#amountPlanned` matches current cart's total amount and currency.
+
+#### Validate payment transaction
+Cart's payment counts as successful if there is at least one payment object with successful (`Payment#Transaction#state`=`Success`) payment transaction of type `Charge` and for the same payment object there is no other successful (`#Payment#Transaction#state`=`Success`) transactions of **other** type than `Charge` and `#Transaction#amount` greater than `0`.
+
+### Possible edge cases
+1. First case:
```

Comment:
let call somehow these cases, for example:
  First case - _delivery address is changed_
  Second case - _cart amount is changed_